src/cargo/sources/registry/remote.rs

```diff
@@ -21,29 +23,76 @@ use std::task::{ready, Poll};
 /// A remote registry is a registry that lives at a remote URL (such as
 /// crates.io). The git index is cloned locally, and `.crate` files are
 /// downloaded as needed and cached locally.
+///
+/// This type is primarily accessed through the [`RegistryData`] trait.
+///
+/// Using Git to host this index provides some benefits. The most prominent one
+/// is that the entire index can be stored efficiently locally on disk. This
```

Comment:
> stored efficiently locally on disk

The on-disk format is the same.

I think you are instead wanting to focus on having the full index available locally with efficient incremental updates ... until we needed to squash the index (and is a pain for initial downloads)